pandas/core/groupby/groupby.py

```diff
@@ -2015,10 +2015,16 @@ def _transform(self, func, *args, engine=None, engine_kwargs=None, **kwargs):
             with com.temp_setattr(self, "as_index", True):
                 # GH#49834 - result needs groups in the index for
                 # _wrap_transform_fast_result
-                if engine is not None:
-                    kwargs["engine"] = engine
-                    kwargs["engine_kwargs"] = engine_kwargs
-                result = getattr(self, func)(*args, **kwargs)
+                if func in ["idxmin", "idxmax"]:
```

Comment:
Is it not possible to stick with the same engine pattern that is in place for this? At first glance I'm wondering what makes these different from `min`/`max`` that requires branching here

Reply:
i think this is the same issue as i asked about somewhere else: _idxmax_idxmin accepts ignore_unobserved while idxmin/idxmax do not

Reply:
The main difference in behavior is that min/max do not raise on unobserved categories, while idxmin and idxmax do.

<details>
<summary>Example - agg</summary>

```
df = pd.DataFrame({'a': pd.Categorical([1, 1, 2], categories=[1, 2, 3]), 'b': [3, 4, 5]})
gb = df.groupby('a', observed=False)
result = gb.min()
print(result)
#      b
# a     
# 1  3.0
# 2  5.0
# 3  NaN
```

</details>

However, the fact that we don't do something special for min/max means that transform unnecessarily coerces to float:

<details>
<summary>Example - transform</summary>

```
df = pd.DataFrame({'a': pd.Categorical([1, 1, 2], categories=[1, 2, 3]), 'b': [3, 4, 5]})
gb = df.groupby('a', observed=False)
result = gb.transform('min')
print(result)
#      b
# 0  3.0
# 1  3.0
# 2  5.0
```

</details>

I consider this a bug in min/max with transform.

Reply:
I've opened #55326 to track

Reply:
Ah OK thanks - that is helpful context. So do you see the solution for min/max being the same as what you have for idxmin/idxmax here? 

I think the broader issue is that we've wanted over time to move away from branching for function specializations within groupby. If that still holds true then I wonder what prevents us from sticking with the existing kwargs interface to solve both this PR and eventually solve min/max's issue

Reply:
I'd like to explore different solutions for min/max and other aggregations, but I don't know what that could be at this time.

I don't understand what you're suggesting with kwargs.

Reply:
Sorry I thought the engine / engine_kwargs were specialized arguments for each function implementation, but I see now those are meant for numba.

The numba functions are UDFs right? I'm assuming from the branch here that we would never want to pass numba arguments to `_idxmax_idxmin`

Reply:
Correct - we never get here when using numba.